src/winmenu.c

```diff
@@ -71,14 +71,19 @@ Menu *CreateWindowMenu(ClientNode *np)
       AddWindowMenuItem(menu, _("Minimize"), MA_MINIMIZE, np, 0);
    }
 
-   if(!(np->state.status & STAT_FULLSCREEN)) {
+   if(!(np->state.status & STAT_FULLSCREEN) && 
+      !(np->state.status & STAT_MINIMIZED)  && 
+       (np->state.border & BORDER_MIN))
+   {
       if(np->state.status & STAT_SHADED) {
          AddWindowMenuItem(menu, _("Unshade"), MA_SHADE, np, 0);
       } else if(np->state.border & BORDER_SHADE) {
          AddWindowMenuItem(menu, _("Shade"), MA_SHADE, np, 0);
       }
       if((np->state.border & BORDER_MAX) &&
-         (np->state.status & (STAT_MAPPED | STAT_SHADED))) {
+         (np->state.status & STAT_MAPPED) && 
+        !(np->state.status & STAT_SHADED))
+      {
```

Comment:
A shaded client is not mapped, so I think simply removing `STAT_SHADED` should give the same effect.  I’m not sure about this change though.  It’s possible to shade/unshade a maximized client, so why not make it possible to maximize/restore a shaded client?